### PR TITLE
Improve (#356) to support registration of wildcard URLs

### PR DESCRIPTION
**Pull request by coggsfl**
Improve on (#356) issue to support wildcard URLs required by Azure API path structure
Add Azure ChatCompletions test cases including test of custom deployment name

### CODE REVIEW

**Comment by coggsfl** [comment]
@sashabaranov please review.  I have a larger PR I'm working on for Azure DALL-E 2 support, however this is a simple change which allows test support for Azure OpenAI endpoints.

**Comment by coggsfl** [comment]
@sashabaranov let me know if you have any questions about the need for wildcard support in the test handlers?  

If the project is going to provide first class support for Azure OpenAI then support for wildcard handlers is required due to the deployment name being a component of the URL structure : /openai/deployments/{model}/chat/completions?api-version={api_version}

File: chat_test.go [hunk 1]
```diff
@@ -67,6 +67,42 @@ func TestChatCompletions(t *testing.T) {
...
+		Model:     GPT3Dot5Turbo,
+		Messages: []ChatCompletionMessage{
+			{
+				Role:    ChatMessageRoleUser,
+				Content: "Hello!",
+			},
+		},
+	})
+	checks.NoError(t, err, "CreateAzureChatCompletionWithCustomDeploymentName error")
+}
```
Comment (vvatanabe) [inline chat_test.go:104]: @coggsfl Why is this test case necessary? The following test case seems to cover it.
https://github.com/sashabaranov/go-openai/blob/master/config_test.go#L9-L62

Comment (coggsfl) [reply chat_test.go:104]: Good call @vvatanabe!  I'll pull that one back.

**Comment by vvatanabe** [comment]
LGTM

**Review by sashabaranov — APPROVED** [review]
Thank you for the PR!

**Pull request merged**